### PR TITLE
add some basic math functor for SelectedRows

### PR DESCRIPTION
**Pull request by QiJune**

### CODE REVIEW

File: paddle/operators/math/math_function.cc [hunk 1]
```diff
@@ -130,6 +130,65 @@ void matmul<platform::CPUPlace, double>(
       matrix_b.data<double>(), beta, matrix_out->data<double>());
 }
 
+template struct SetConstant<platform::CPUPlace, float>;
+
+namespace detail {
+size_t FindPos(const std::vector<int64_t>& rows, int64_t value) {
```
Comment (reyoung) [inline paddle/operators/math/math_function.cc]: return std::find(rows.begin(), rows.end(), value) - rows.begin();

File: paddle/operators/math/math_function.cc [hunk 2]
```diff
@@ -130,6 +132,99 @@ void matmul<platform::CPUPlace, double>(
       matrix_b.data<double>(), beta, matrix_out->data<double>());
 }
 
+template struct SetConstant<platform::CPUPlace, float>;
```
Comment (wangkuiyi) [inline paddle/operators/math/math_function.cc]: Let us move the added code into `paddle/operators/math/selected_rows.{h,cc,cu}`.

File: paddle/operators/math/selected_rows_functor.cc [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+    auto in1_place = input1.place();
+    PADDLE_ENFORCE(platform::is_cpu_place(in1_place));
+    auto in2_place = input2.place();
+    PADDLE_ENFORCE(platform::is_cpu_place(in2_place));
+    auto out_place = context.GetPlace();
+    PADDLE_ENFORCE(platform::is_cpu_place(out_place));
+
+    auto* out_data = out_value->data<T>();
+    auto* in1_data = in1_value.data<T>();
+    memory::Copy(boost::get<platform::CPUPlace>(out_place), out_data,
```
Comment (typhoonzero) [inline paddle/operators/math/selected_rows_functor.cc]: I'm not sure, if we have duplicated rows selected, the row value is copied multiple times?

File: paddle/operators/math/math_function.cc (hunk 2 above)
Comment (QiJune) [reply paddle/operators/math/math_function.cc]: Done

File: paddle/operators/math/selected_rows_functor.cc (hunk 3 above)
Comment (QiJune) [reply paddle/operators/math/selected_rows_functor.cc]: Yes, the row value will be just simply concated.
```
// concat rows
  out_rows.insert(out_rows.end(), in1_rows.begin(), in1_rows.end());
  out_rows.insert(out_rows.end(), in2_rows.begin(), in2_rows.end());
  output->set_rows(out_rows);
```

**Pull request merged**